src/request.rs

```diff
@@ -1,119 +1,24 @@
-#[cfg(feature = "http")]
-extern crate reqwest;
-
-use error::{Error, RequestErrorKind};
-use parser::parse_response;
-use transport::Transport;
-use utils::escape_xml;
-use Value;
-
-use std::collections::BTreeMap;
 use std::io::{self, Write};
+use xml::escape::{escape_str_pcdata as escape_xml};
+
+use crate::Value;
 
 /// A request to call a procedure.
 #[derive(Clone, Debug)]
 pub struct Request<'a> {
     name: &'a str,
-    args: Vec<Value>,
+    args: &'a [Value],
 }
 
 impl<'a> Request<'a> {
     /// Creates a new request to call a function named `name`.
-    ///
-    /// By default, no arguments are passed. Use the `arg` method to append arguments.
-    pub fn new(name: &'a str) -> Self {
+    pub fn new(name: &'a str, args: &'a [Value]) -> Self {
```

Comment:
If you did the above comment then you could force a move of the `Vec<Value>` here